webrtc/README.md

```diff
@@ -0,0 +1,445 @@
+        - [Head-of-line blocking](#head-of-line-blocking)
+        - [`RTCDataChannel` negotiation](#rtcdatachannel-negotiation)
+        - [`RTCDataChannel` label](#rtcdatachannel-label)
+    - [Connection Security](#connection-security)
+    - [Previous, ongoing and related work](#previous-ongoing-and-related-work)
+- [FAQ](#faq)
+
+<!-- markdown-toc end -->
+
+## Motivation
+
+1. **No need for trusted TLS certificates.** Enable browsers to connect to
+   public server nodes without those server nodes providing a TLS certificate
+   within the browsers trustchain. Note that we can not do this today with our
+   Websocket transport as the browser requires the remote to have a trusted TLS
+   certificate. Nor can we establish a plain TCP or QUIC connection from within
+   a browser.
+
+## Addressing
+
+WebRTC multiaddresses are composed of an IP and UDP address component, followed
+by `/webrtc` and a multihash of the certificate that the node uses.
+
+Examples:
+- `/ip4/1.2.3.4/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>`
+- `/ip6/fe80::1ff:fe23:4567:890a/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>`
+
+The TLS certificate fingerprint in `/certhash` is a
+[multibase](https://github.com/multiformats/multibase) encoded
+[multihash](https://github.com/multiformats/multihash).
+
+For compatibility implementations MUST support hash algorithm
+[`sha-256`](https://github.com/multiformats/multihash) and base encoding
+[`base64url`](https://github.com/multiformats/multibase). Implementations MAY
+support other hash algorithms and base encodings, but they may not be able to
+connect to all other nodes.
+
+## Connection Establishment
+
+### Browser to public Server
+
+Scenario: Browser _A_ wants to connect to server node _B_ where _B_ is publicly
+reachable but _B_ does not have a TLS certificate trusted by _A_.
+
+1. Server node _B_ generates a TLS certificate, listens on a UDP port and
+   advertises the corresponding multiaddress (see [#Addressing]) through some
+   external mechanism.
+
+   Given that _B_ is publicly reachable, _B_ acts as a [ICE
+   Lite](https://www.rfc-editor.org/rfc/rfc5245) agent. It listens on a UDP port
+   for incoming STUN and SCTP packets and multiplexes based on source IP and
+   source port.
+
+2. Browser _A_ discovers server node _B_'s multiaddr, containing _B_'s IP, UDP
+  port, TLS certificate fingerprint and libp2p peer ID (e.g.
+  `/ip6/2001:db8::/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>`),
+  through some external mechanism.
+
+3. _A_ instantiates a `RTCPeerConnection`. See
+   [`RTCPeerConnection()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/RTCPeerConnection).
+
+   _A_ (i.e. the browser) SHOULD NOT reuse the same certificate across
+   `RTCPeerConnection`s. Reusing the certificate can be used to identify _A_
+   across connections by on-path observers given that WebRTC uses TLS 1.2.
+
+4. _A_ constructs _B_'s SDP offer locally based on _B_'s multiaddr. _A_
```

Comment:
```suggestion
4. _A_ constructs _B_'s SDP answer locally based on _B_'s multiaddr. _A_
```

Reply:
_A_ is the initiator and will create the offer and infer an answer. _B_ will infer an offer based on incoming STUN and create an answer.

Reply:
Thanks @ckousik. Fixed with 1e3ca59.